crates/bevy_pbr/src/render/light.rs

```diff
@@ -1727,10 +1730,11 @@ impl PhaseItem for Shadow {
 
     #[inline]
     fn sort(items: &mut [Self]) {
-        // The shadow phase is sorted by pipeline id for performance reasons.
+        // The shadow phase is sorted by pipeline id and mesh id for performance reasons.
         // Grouping all draw commands using the same pipeline together performs
-        // better than rebinding everything at a high rate.
-        radsort::sort_by_key(items, |item| item.sort_key());
+        // better than rebinding everything at a high rate, and grouping by mesh
+        // allows batching to reduce draw calls.
+        items.sort_by_key(Self::sort_key);
```

Comment:
Can we use `sort_unstable_by_key` here? I found it to be much faster, and I'm not sure if stable sorting is required (if it is then I'll need to make a PR to fix my opaque pass pipeline sorting PR).

Reply:
👍